### PR TITLE
Definitions and examples for wine characteristics (DENNER-2607)

### PR DESCRIPTION
**Pull request by curzio-della-santa**

### CODE REVIEW

File: src/swagger.yml [hunk 1]
```diff
@@ -1361,6 +1467,29 @@ definitions:
...
+        $ref: '#/definitions/Translations'
+      sort_index:
+        $ref: '#/definitions/SortIndex'
+
+  FlavorScore:
+    type: object
+    properties:
+      id:
+        type: string
+        description: Flsvor ID
```
Comment (curzio-della-santa) [inline src/swagger.yml]: typo

File: examples/wine-flavors.get.json [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                },
+                {
+                    "locale": "it",
+                    "value": "intensità"
+                }
+            ],
+            "sort_index": 3
+        },
+        {
+            "id": "barrique-aging",
```
Comment (curzio-della-santa) [inline examples/wine-flavors.get.json]: @ivan-wolf In the first version, Denner called this flavor "Veredlung" .. that would translate to "finishing"  

Aged in barrique barrel is also called "barricaded" ( "verbarrikadiert" in german)

What do you think is the best naming for this?

File: examples/wine-flavors.get.json [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                    "locale": "de",
+                    "value": "Barrique Ausbau"
+                },
+                {
+                    "locale": "fr",
+                    "value": "..."
+                },
+                {
+                    "locale": "it",
+                    "value": "barricato"
```
Comment (curzio-della-santa) [inline examples/wine-flavors.get.json]: Is probably not correct, this translates in german to "verbarrikadiert"

File: examples/wine-flavors.get.json [hunk 4]
```diff
@@ -0,0 +1,95 @@
+{
+    "wine_flavors": [
```
Comment (curzio-della-santa) [inline examples/wine-flavors.get.json]: I've searched a bit, "flavor" might not be the correct wording, I suggest to change wording to "**taste**" or "characteristics"

Ref: https://winefolly.com/deep-dive/wine-characteristics/

Comment (ivan-wolf) [reply examples/wine-flavors.get.json]: The term "wine characteristics" seems appropriate.

File: examples/wine-flavors.get.json [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                },
+                {
+                    "locale": "it",
+                    "value": "acidità"
+                }
+            ],
+            "sort_index": 1
+        },
+        {
+            "id": "sweet",
```
Comment (ivan-wolf) [inline examples/wine-flavors.get.json]: `sweetness`

File: examples/wine-flavors.get.json [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                }
+            ],
+            "sort_index": 3
+        },
+        {
+            "id": "barrique-aging",
+            "titles": [
+                {
+                    "locale": "de",
+                    "value": "Barrique Ausbau"
```
Comment (ivan-wolf) [inline examples/wine-flavors.get.json]: `Barrique-Ausbau`

File: examples/wine-flavors.get.json (hunk 2 above)
Comment (ivan-wolf) [reply examples/wine-flavors.get.json]: `barrique-aging` seems fine.

File: examples/wines.get.json [hunk 7]
```diff
@@ -282,6 +282,28 @@
                         "score": 1
                     }
                 ],
+                "flavors": [
+                    {
+                        "id": "acid",
+                        "score": 1
```
Comment (ivan-wolf) [inline examples/wines.get.json]: For food matches wie include the titles. But there are benefits to not include it (we could set up the model `WineCharacteristic` without reference to `WineCharacteristicScores`. Should we go with this approach, I recommend renaming `id` to `characteristic` (and maybe `score` to `value`).

File: src/swagger.yml [hunk 8]
```diff
@@ -708,6 +724,94 @@ paths:
...
 
+  ### Wine flavors
+  /wine-flavors:
+    get:
+      summary: List of wine flavors
+      description: >
+        Use this operation to get a list of all wine flavors (Wein-Geschmacksprofil).
+      tags:
+        - wineFlavor
+      operationId: listWineGrowers
```
Comment (ivan-wolf) [inline src/swagger.yml]: Modify accordingly.

**Review by ivan-wolf — CHANGES_REQUESTED** [review]

File: examples/wines.get.json (hunk 7 above)
Comment (curzio-della-santa) [reply examples/wines.get.json]: The on top `flavor` will be replaced with `characteristic` .. therefore `id` has to remain.

File: examples/wines.get.json [hunk 9]
```diff
@@ -708,11 +730,11 @@
                         },
                         {
                             "locale": "fr",
-                            "value": "<ul><li>​Allgemeiner <i>Hinweis</i></li></ul>"
+                            "value": "<ul><li>Allgemeiner <i>Hinweis</i></li></ul>"
```
Comment (curzio-della-santa) [inline examples/wines.get.json:733]: Unrelated fix: Removed a zero-width space character between `<li>` and `Allgemeiner` .. same for the coming changes

**Comment by curzio-della-santa** [comment]
Should change branch name, but I'm not sure if then al the review will be lost ..

File: examples/wines.get.json (hunk 7 above)
Comment (ivan-wolf) [reply examples/wines.get.json]: > therefore `id` has to remain.

It has to do how we set up the model, and not wether it's called `flavor` or `characteristic`.
The data here represents `WineCharacteristicScore` for which I suggest `characteristic` and `value` as properties.

Comment (ivan-wolf) [reply examples/wines.get.json]: See this draft for details: https://bitbucket.org/detailnet/denner-common/pull-requests/228/add-models-for-wine-characteristics-denner

File: src/swagger.yml [hunk 10]
```diff
@@ -2057,7 +2221,7 @@ definitions:
         description: The number of items on all pages
         type: integer
         format: int32
-      wine_growers:
+      wine_experts:
```
Comment (curzio-della-santa) [inline src/swagger.yml:2224]: Unrelated fix

File: examples/wines.get.json (hunk 7 above)
Comment (curzio-della-santa) [reply examples/wines.get.json]: 👍  done

File: examples/wines.get.json [hunk 11]
```diff
@@ -282,6 +282,28 @@
                         "score": 1
                     }
                 ],
+                "characteristics": [
```
Comment (ivan-wolf) [inline examples/wines.get.json]: @curzio-della-santa, what do you think about `characteristic_scores` to make it more clear?

Comment (curzio-della-santa) [reply examples/wines.get.json]: ok
